src/utils/mutate.ts

```diff
@@ -22,7 +22,9 @@ export const internalMutate = async <Data>(
     typeof _opts === 'boolean' ? { revalidate: _opts } : _opts || {}
 
   // Fallback to `true` if it's not explicitly set to `false`
-  let populateCache = options.populateCache !== false
+  let populateCache = isUndefined(options.populateCache)
+    ? true
```

Comment:
default populateCache can be a function then we don't need to check it later

```suggestion
    ? (v => v)
```